src/session/session.controller.ts

```diff
@@ -0,0 +1,117 @@
+import { ParseMongoIdPipe } from '@/pipes/mongo-id.pipe';
+import { SessionSchema } from '@/schema/session.schema';
+import { CreateSessionDto } from '@/session/dto/create-session.dto';
+import { SessionService } from '@/session/session.service';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpException,
+  HttpStatus,
+  Param,
+  ParseIntPipe,
+  Patch,
+  Post,
+  Query,
+} from '@nestjs/common';
+import { ApiParam, ApiQuery, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { MongooseError, Types } from 'mongoose';
+
+@Controller('sessions')
+@ApiTags('sessions')
+export class SessionController {
+  constructor(private readonly sessionService: SessionService) {}
+
+  @ApiQuery({ name: 'finished', type: Boolean, required: false })
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'Sessions list',
+    type: () => SessionSchema,
+    isArray: true,
+  })
+  @Get()
+  getSessions(@Query('finished') finished?: boolean) {
+    return this.sessionService.getSessions(finished);
+  }
```

Comment:
Would be nice if it can be filtered by date, for example `?before=2023‐07‐29T23:28:12−07:00`.

Or not? Maybe, this needs some of your opinions as well,